src/mon/MonCap.cc

```diff
@@ -601,14 +601,12 @@ struct MonCapParser : qi::grammar<Iterator, MonCap()>
 
 bool MonCap::parse(const string& str, ostream *err)
 {
-  string s = str;
-  string::iterator iter = s.begin();
-  string::iterator end = s.end();
-
-  MonCapParser<string::iterator> g;
-  bool r = qi::parse(iter, end, g, *this);
-  //MonCapGrant foo;
-  //bool r = qi::phrase_parse(iter, end, g, ascii::space, foo);
+  auto iter = str.begin();
+  auto end = str.end();
+
+  MonCapParser<string::const_iterator> pexp;
```

Comment:
why you are switching to `const_iterator` here?

Reply:
Because it is needed by `phrase_parse`.

Reply:
i think the reason is that the `str` here is a `const`, and we can only feed the `boost::spirit::qi::parse()` with `const_iterator`, hence `expr` should be a `qi::grammar<const_iterator,..>`.